File: day1/day1.cpp
#include <iostream>
#include <vector>
#include <string>
#include <fstream>
#include <sstream>
#include <bits/stdc++.h>
using namespace std;

void createCalorieVector(vector<int>& calories, string filename)
{
    string input = "";
    int counter = 0;
    ifstream readFromFile(filename);
    while (getline(readFromFile, input)) {
        if (input != "")
            counter += stoi(input);
        if (input == "") {
            calories.push_back(counter);
            counter = 0;
        }
    }

}

void calculateCalories(vector<int>& calories)
{
    sort(calories.begin(), calories.end(), greater<int>());
    cout << "Max Calories: " << calories[0] << endl;
    cout << "Sum of Top Three Calories: " << calories[0] + calories[1] + calories[2] << endl;
}

int main()
{
    vector<int> large_input_calories;
    vector<int> small_input_calories;

    //Create vector of calories for each input
    createCalorieVector(small_input_calories, "small_input.txt");
    createCalorieVector(large_input_calories, "input_day1.txt");

    //Calculate and print the 
    calculateCalories(small_input_calories);
    calculateCalories(large_input_calories);

    //For Debug
//    printVector(small_input_calories);
//    printVector(large_input_calories);

    return 0;
}

File: day1/day1.py

def getCalorieList(file_path):
    with open(file_path, 'r') as file:
        calorie_list = file.read().splitlines()
    return calorie_list

def getSumPerElfCalorieList(calorie_list):
    counter = 0
    condensed_calorie_list = []
    for i in range(len(calorie_list)):
        if calorie_list[i] != "":
            counter += int(calorie_list[i])
        if calorie_list[i] == "":
            condensed_calorie_list.append(counter)
            counter = 0
    return sorted(condensed_calorie_list, reverse=True)

def printCalories(calorie_list):
    print("Max Calories: " + str(calorie_list[0]))
    print("Sum of Top Three Calories: " + str(calorie_list[0] + calorie_list[1] + calorie_list[2]))

def main():
    small_calorie_list = getCalorieList("small_input.txt")
    large_calorie_list = getCalorieList("input_day1.txt")

    condensed_small_list = getSumPerElfCalorieList(small_calorie_list)
    condensed_large_list = getSumPerElfCalorieList(large_calorie_list)

    print("Day 1 Part 1")
    printCalories(condensed_small_list)
    print()
    print("Day 1 Part 2")
    printCalories(condensed_large_list)

    #For debug
#    print(small_calorie_list)
#    print(large_calorie_list)

if __name__ == "__main__":
    main()